src/overlays/actors/ovl_En_Wiz_Fire/z_en_wiz_fire.c

```diff
@@ -35,51 +51,838 @@ const ActorInit En_Wiz_Fire_InitVars = {
+                this->unk_162 = 0;
+                this->collider.info.toucher.damage = 8;
+            }
+
+        case 1:
+        case 3:
+            this->actionFunc = func_80A4984C;
+            break;
+
+        case 2:
+            this->actor.draw = func_80A4BAB4;
+            this->unk_170 = Rand_S16Offset(0, 10000);
+            this->unk_160 = 1;
+            this->collider.info.toucher.damage = 2;
+            this->actionFunc = func_80A49F38;
+    }
+}
+
+void EnWizFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnWizFire* this = THIS;
+
+    if (this->unk_162 == 0) {
+        globalCtx->envCtx.unk_8C.fogColor[2] = 0;
+        globalCtx->envCtx.unk_8C.fogColor[1] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.fogColor[0] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor2[2] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor2[1] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor2[0] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor1[2] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor1[1] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.diffuseColor1[0] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.ambientColor[2] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.ambientColor[1] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.ambientColor[0] = globalCtx->envCtx.unk_8C.fogColor[2];
+        globalCtx->envCtx.unk_8C.fogNear = globalCtx->envCtx.unk_8C.fogColor[2];
+    }
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80A4984C(EnWizFire* this, GlobalContext* globalCtx) {
+    Vec3f sp44 = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_178); i++) {
+        this->unk_178[i] = this->actor.world.pos;
+    }
+
+    this->unk_16E = 0;
+    Matrix_StatePush();
+    Matrix_RotateY(this->actor.world.rot.y, MTXMODE_NEW);
+    Matrix_InsertXRotation_s(this->actor.world.rot.x, MTXMODE_APPLY);
+
+    if (this->unk_162 != 0) {
+        sp44.z = randPlusMinusPoint5Scaled(2.0f) + 8.0f;
+    } else {
+        sp44.z = 12.0f;
+    }
+
+    Matrix_MultiplyVector3fByState(&sp44, &this->actor.velocity);
+    Matrix_StatePop();
+    this->actor.world.rot.x = this->actor.world.rot.y = this->actor.world.rot.z = 0;
+    this->unk_168 = 50;
+    if (this->unk_162 != 0) {
+        this->actor.velocity.y = 10.0f;
+        this->actor.gravity = -1.0f;
+        this->unk_150 = 0.01f;
+    } else {
+        this->unk_150 = 0.02f;
+        this->unk_168 = 100;
+    }
+    this->unk_160 = 0;
+    this->actionFunc = func_80A49A44;
+}
+
+void func_80A49A44(EnWizFire* this, GlobalContext* globalCtx) {
+    Vec3f sp54 = { 0.0f, 0.0f, 0.0f };
+
+    this->actor.world.rot.z += 0x1388;
```

Comment:
This is a round number
```suggestion
    this->actor.world.rot.z += 5000;
```